File: src/main/java/com/will/exception/VisionagentException.java
package com.will.exception;

import com.will.constant.ExceptionCode;

/**
 * Created by Will on 2016/8/17 9:11.
 */
public class VisionagentException extends RuntimeException {

    private ExceptionCode code;

    public VisionagentException(ExceptionCode code) {
        this.code = code;
    }

    public VisionagentException(ExceptionCode code, String message) {
        super(message);
        this.code = code;
    }

    public VisionagentException(ExceptionCode code, String message, Throwable cause) {
        super(message, cause);
        this.code = code;
    }

    public VisionagentException(ExceptionCode code, Throwable cause) {
        super(cause);
        this.code = code;
    }

    public VisionagentException(ExceptionCode code, String message, Throwable cause, boolean enableSuppression, boolean writableStackTrace) {
        super(message, cause, enableSuppression, writableStackTrace);
        this.code = code;
    }

    @Override
    public String toString() {
        String str = "错误码：" + code + " " + getLocalizedMessage();
        return str;
    }
}

File: src/main/java/com/will/exception/VisionagentException.kt
package com.will.exception

import com.will.constant.ExceptionCode
import com.will.utils.Value

/**
 * Created by zoumy on 2017/5/11 10:59.
 */
class VisionagentException : RuntimeException {
    val code:ExceptionCode

    constructor(code: ExceptionCode) : super() {
        this.code = code
    }

    constructor(message: String?, code: ExceptionCode) : super(message) {
        this.code = code
    }

    constructor(message: String?, cause: Throwable?, code: ExceptionCode) : super(message, cause) {
        this.code = code
    }

    constructor(cause: Throwable?, code: ExceptionCode) : super(cause) {
        this.code = code
    }

    constructor(message: String?, cause: Throwable?, enableSuppression: Boolean, writableStackTrace: Boolean, code: ExceptionCode) : super(message, cause, enableSuppression, writableStackTrace) {
        this.code = code
    }


    override fun toString(): String {
        return "错误码：$code \n ${Value.ofEmpty(localizedMessage)}"
    }
}